src/wasm/wasm-binary.cpp

```diff
@@ -1088,6 +1088,40 @@ void WasmBinaryWriter::writeDylinkSection() {
   finishSection(start);
 }
 
+void WasmBinaryWriter::writeDylinkSection() {
+  if (!wasm->dylinkSection) {
+    return;
+  }
+
+  if (wasm->dylinkSection->isLegacy) {
```

Comment:
Is this just so legacy dylink sections round-trip?
I wonder if that's actually necessary?

Reply:
It will be needed/useful in the interim period to fully support both formats.   That way we can land this now either before of after the emscripten change.

Reply:
(otherwise running finalize will convert the old to the new format... which isn't a huge problem but it seems better to leave it alone).